src/vs/workbench/api/browser/mainThreadEditors.ts

```diff
@@ -274,6 +276,52 @@ export class MainThreadTextEditors implements MainThreadTextEditorsShape {
 
 		return Promise.resolve([]);
 	}
+
+	$tryGetEncoding(id: string): Promise<string | undefined> {
+		const editor = this._editorLocator.getEditor(id);
+
+		if (!editor) {
+			return Promise.reject(new Error('No such TextEditor'));
+		}
+
+		const editorPanes = this._editorService.visibleEditorPanes;
+		for (const editorPane of editorPanes) {
+			if (editor.matches(editorPane)) {
+				const encodingSupport = toEditorWithEncodingSupport(editorPane.input);
+				if (!encodingSupport) {
+					return Promise.reject(new Error('No file active at this time'));
+				}
+
+				const encoding = encodingSupport.getEncoding();
+
+				return Promise.resolve(encoding);
+			}
+		}
+
+		return Promise.reject(new Error('No text editor active at this time'));
+	}
+
+	$trySetEncoding(id: string, encoding: string): Promise<void> {
```

Comment:
I think the EncodingMode for `encodingSupport.setEncoding` shouldn't be hardcoded but rather be a function parameter.

I created an extension which sets the encoding depending on the file contents. After opening an existing file previously saved with "Windows-1251" the contents of the file completely broke. Changing the EncodingMode here to `EncodingMode.Decode` fixed the issue.

I believe it also makes sense after looking at these references:
https://github.com/microsoft/vscode/blob/53ed07606e51c2047564d8cefebc36a177fa3e6f/src/vs/workbench/browser/parts/editor/editorStatus.ts#L1521
https://github.com/microsoft/vscode/blob/53ed07606e51c2047564d8cefebc36a177fa3e6f/src/vs/workbench/services/textfile/common/textFileEditorModel.ts#L1116